tests/jquery.flot.navigate.Test.js

```diff
@@ -509,7 +397,206 @@ describe("flot navigate plugin", function () {
             expect(yaxis.min).toBe(-10);
             expect(yaxis.max).toBe(0);
         });
+    });
 
+    describe('touchZoom', function() {
+
+        function getDistance(coords) {
+                return Math.sqrt((coords[0].x - coords[1].x) * (coords[0].x - coords[1].x) + ((coords[0].y - coords[1].y) * (coords[0].y - coords[1].y)));
+            }
+
+        it('should zoom the plot',function() {
+            plot = $.plot(placeholder, [
+                [
+                    [-1, 2],
+                    [11, 12]
+                ]
+            ], options);
+
+            var canvasElement = placeholder[0].childNodes[2],
+                xaxis = plot.getXAxes()[0],
+                yaxis = plot.getYAxes()[0],
+                initialXmin = xaxis.min,
+                initialXmax = xaxis.max,
+                initialYmin = yaxis.min,
+                initialYmax = yaxis.max,
+                initialCoords = [
+                    getPairOfCoords(xaxis, yaxis, 3, 5),
+                    getPairOfCoords(xaxis, yaxis, 7, 9)
+                ],
+                finalCoords = [
+                    getPairOfCoords(xaxis, yaxis, 2, 4),
+                    getPairOfCoords(xaxis, yaxis, 8, 10)
+                ],
+                midPointCoords = {
+                        x: (xaxis.c2p(finalCoords[0].x - plot.offset().left) + xaxis.c2p(finalCoords[1].x - plot.offset().left)) / 2,
+                        y: (yaxis.c2p(finalCoords[0].y - plot.offset().top) + yaxis.c2p(finalCoords[1].y - plot.offset().top)) / 2
+                },
+                amount = getDistance(finalCoords) / getDistance(initialCoords);
+
+            simulate.sendTouchEvents(initialCoords, placeholder[0].childNodes[2], "touchstart");
+            simulate.sendTouchEvents(finalCoords, placeholder[0].childNodes[2], "touchmove");
+            simulate.sendTouchEvents(finalCoords, placeholder[0].childNodes[2], "touchend");
+
+            expect(xaxis.min).toBeCloseTo((midPointCoords.x - initialXmin) * (1 - 1/amount) + initialXmin, 6);
+            expect(xaxis.max).toBeCloseTo(initialXmax - (initialXmax - midPointCoords.x) * (1 - 1/amount), 6);
+            expect(yaxis.min).toBeCloseTo((midPointCoords.y - initialYmin) * (1 - 1/amount) + initialYmin, 6);
+            expect(yaxis.max).toBeCloseTo(initialYmax - (initialYmax - midPointCoords.y) * (1 - 1/amount), 6);
+          });
+
+        it('should zoom the plot correctly using pageXY when the canvas is placed in the bottom scrollable area of the page', function () {
+              var newDiv = $('<div style="height: 800px"> hello </div>');
+              $(newDiv).insertBefore('#test-container');
```

Comment:
Is it possible to reuse the placeholder variable? In case the #test-container id changes you don't want to change it all over the places.